pkg/asset/kubeconfig/kubeconfig_test.go

```diff
@@ -0,0 +1,246 @@
+package kubeconfig
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+type fakeAsset int
+
+var _ asset.Asset = fakeAsset(0)
+
+func (f fakeAsset) Dependencies() []asset.Asset {
+	return nil
+}
+
+func (f fakeAsset) Generate(map[asset.Asset]*asset.State) (*asset.State, error) {
+	return nil, nil
+}
+
+func TestKubeconfigGenerate(t *testing.T) {
+	testDir, err := ioutil.TempDir(os.TempDir(), "kubeconfig_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(testDir)
+
+	rootCA := fakeAsset(0)
+	adminCertKey := fakeAsset(1)
+	kubeletCertKey := fakeAsset(2)
+	installConfig := fakeAsset(3)
+
+	rootCAState := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: "root-ca.key",
+				Data: []byte("THIS IS ROOT CA KEY DATA"),
+			},
+			{
+				Name: "root-ca.crt",
+				Data: []byte("THIS IS ROOT CA CERT DATA"),
+			},
+		},
+	}
+
+	adminCertState := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: "admin.key",
+				Data: []byte("THIS IS ADMIN KEY DATA"),
+			},
+			{
+				Name: "admin.crt",
+				Data: []byte("THIS IS ADMIN CERT DATA"),
+			},
+		},
+	}
+
+	kubeletCertState := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: "kubelet.key",
+				Data: []byte("THIS IS KUBELET KEY DATA"),
+			},
+			{
+				Name: "kubelet.crt",
+				Data: []byte("THIS IS KUBELET CERT DATA"),
+			},
+		},
+	}
+
+	installConfigState := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: "installconfig.yaml",
+				Data: []byte(`admin:
+  email: foo
+  password: foo
+baseDomain: test.base.domain
+clusterID: e93844a3-a727-11e8-ab80-28d244b00276
+license: foo
+machines:
+metadata:
+  creationTimestamp: null
+  name: test-cluster-name
+networking:
+  podCIDR:
+    IP: ""
+    Mask: null
+  serviceCIDR:
+    IP: ""
+    Mask: null
+  type: ""
+platform:
+  aws:
+    keyPairName: foo
+    region: us-west-2
+    vpcCIDRBlock: ""
+    vpcID: ""
+pullSecret: foo
+`),
+			},
+		},
+	}
+
+	tests := []struct {
+		userName     string
+		certKey      asset.Asset
+		parents      map[asset.Asset]*asset.State
+		err          bool
```

Comment:
nit: these tests would be more robust if you used a regex string here, and then used [`assert.Regexp`][1], [`assert.EqualErrorf`][2], or something like that.  That would give us failing tests when something unexpected broke (vs. getting the expected error).

[1]: https://godoc.org/github.com/stretchr/testify/assert#Regexp
[2]: https://godoc.org/github.com/stretchr/testify/assert#EqualErrorf